{"nbformat":4,"nbformat_minor":0,"metadata":{"colab":{"provenance":[],"authorship_tag":"ABX9TyOlw6HIeFyIraJsZj80eeIz"},"kernelspec":{"name":"python3","display_name":"Python 3"},"language_info":{"name":"python"}},"cells":[{"cell_type":"code","execution_count":1,"metadata":{"colab":{"base_uri":"https://localhost:8080/"},"id":"rm0ipjzC3EUt","executionInfo":{"status":"ok","timestamp":1668441552152,"user_tz":300,"elapsed":3448,"user":{"displayName":"Francesco Bassino","userId":"12214195385968794219"}},"outputId":"e8fb3add-0145-458d-8581-538b54a1a798"},"outputs":[{"output_type":"stream","name":"stdout","text":["Cloning into 'Deep-Generative-Projection'...\n","remote: Enumerating objects: 152, done.\u001b[K\n","remote: Counting objects: 100% (152/152), done.\u001b[K\n","remote: Compressing objects: 100% (138/138), done.\u001b[K\n","remote: Total 152 (delta 33), reused 113 (delta 12), pack-reused 0\u001b[K\n","Receiving objects: 100% (152/152), 31.57 MiB | 13.39 MiB/s, done.\n","Resolving deltas: 100% (33/33), done.\n"]}],"source":["!git clone https://github.com/RuiLiFeng/Deep-Generative-Projection"]},{"cell_type":"code","source":["%%bash\n","conda create -n DGP python=3.6 cudatoolkit=10.0\n","conda activate DGP\n","cd Deep-Generative-Projection/\n","pip install -r requirements.txt"],"metadata":{"colab":{"base_uri":"https://localhost:8080/"},"id":"AMtF6ff83QEj","executionInfo":{"status":"ok","timestamp":1668441671301,"user_tz":300,"elapsed":70305,"user":{"displayName":"Francesco Bassino","userId":"12214195385968794219"}},"outputId":"a3fc852a-d97d-44b2-da44-d5e77c38264f"},"execution_count":2,"outputs":[{"output_type":"stream","name":"stdout","text":["Looking in indexes: https://pypi.org/simple, https://us-python.pkg.dev/colab-wheels/public/simple/\n","Collecting numpy==1.16.4\n","  Downloading numpy-1.16.4-cp37-cp37m-manylinux1_x86_64.whl (17.3 MB)\n","Collecting tensorflow-gpu==1.14.0\n","  Downloading tensorflow_gpu-1.14.0-cp37-cp37m-manylinux1_x86_64.whl (377.1 MB)\n","Collecting opencv-python==4.5.3.56\n","  Downloading opencv_python-4.5.3.56-cp37-cp37m-manylinux2014_x86_64.whl (49.9 MB)\n","Collecting matplotlib==3.3.4\n","  Downloading matplotlib-3.3.4-cp37-cp37m-manylinux1_x86_64.whl (11.5 MB)\n","Collecting Keras==2.2.5\n","  Downloading Keras-2.2.5-py2.py3-none-any.whl (336 kB)\n","Collecting Pillow==8.4.0\n","  Downloading Pillow-8.4.0-cp37-cp37m-manylinux_2_17_x86_64.manylinux2014_x86_64.whl (3.1 MB)\n","Collecting tqdm==4.32.2\n","  Downloading tqdm-4.32.2-py2.py3-none-any.whl (50 kB)\n","Collecting h5py==2.9.0\n","  Downloading h5py-2.9.0-cp37-cp37m-manylinux1_x86_64.whl (2.8 MB)\n","Collecting imageio==2.15.0\n","  Downloading imageio-2.15.0-py3-none-any.whl (3.3 MB)\n","Collecting imageio-ffmpeg==0.4.7\n","  Downloading imageio_ffmpeg-0.4.7-py3-none-manylinux2010_x86_64.whl (26.9 MB)\n","Requirement already satisfied: requests in /usr/local/lib/python3.7/dist-packages (from -r requirements.txt (line 11)) (2.23.0)\n","Requirement already satisfied: six>=1.10.0 in /usr/local/lib/python3.7/dist-packages (from tensorflow-gpu==1.14.0->-r requirements.txt (line 2)) (1.15.0)\n","Collecting tensorflow-estimator<1.15.0rc0,>=1.14.0rc0\n","  Downloading tensorflow_estimator-1.14.0-py2.py3-none-any.whl (488 kB)\n","Requirement already satisfied: google-pasta>=0.1.6 in /usr/local/lib/python3.7/dist-packages (from tensorflow-gpu==1.14.0->-r requirements.txt (line 2)) (0.2.0)\n","Requirement already satisfied: protobuf>=3.6.1 in /usr/local/lib/python3.7/dist-packages (from tensorflow-gpu==1.14.0->-r requirements.txt (line 2)) (3.19.6)\n","Requirement already satisfied: astor>=0.6.0 in /usr/local/lib/python3.7/dist-packages (from tensorflow-gpu==1.14.0->-r requirements.txt (line 2)) (0.8.1)\n","Requirement already satisfied: absl-py>=0.7.0 in /usr/local/lib/python3.7/dist-packages (from tensorflow-gpu==1.14.0->-r requirements.txt (line 2)) (1.3.0)\n","Requirement already satisfied: keras-preprocessing>=1.0.5 in /usr/local/lib/python3.7/dist-packages (from tensorflow-gpu==1.14.0->-r requirements.txt (line 2)) (1.1.2)\n","Requirement already satisfied: wheel>=0.26 in /usr/local/lib/python3.7/dist-packages (from tensorflow-gpu==1.14.0->-r requirements.txt (line 2)) (0.38.3)\n","Requirement already satisfied: gast>=0.2.0 in /usr/local/lib/python3.7/dist-packages (from tensorflow-gpu==1.14.0->-r requirements.txt (line 2)) (0.4.0)\n","Requirement already satisfied: wrapt>=1.11.1 in /usr/local/lib/python3.7/dist-packages (from tensorflow-gpu==1.14.0->-r requirements.txt (line 2)) (1.14.1)\n","Collecting keras-applications>=1.0.6\n","  Downloading Keras_Applications-1.0.8-py3-none-any.whl (50 kB)\n","Collecting tensorboard<1.15.0,>=1.14.0\n","  Downloading tensorboard-1.14.0-py3-none-any.whl (3.1 MB)\n","Requirement already satisfied: termcolor>=1.1.0 in /usr/local/lib/python3.7/dist-packages (from tensorflow-gpu==1.14.0->-r requirements.txt (line 2)) (2.1.0)\n","Requirement already satisfied: grpcio>=1.8.6 in /usr/local/lib/python3.7/dist-packages (from tensorflow-gpu==1.14.0->-r requirements.txt (line 2)) (1.50.0)\n","Requirement already satisfied: pyparsing!=2.0.4,!=2.1.2,!=2.1.6,>=2.0.3 in /usr/local/lib/python3.7/dist-packages (from matplotlib==3.3.4->-r requirements.txt (line 4)) (3.0.9)\n","Requirement already satisfied: python-dateutil>=2.1 in /usr/local/lib/python3.7/dist-packages (from matplotlib==3.3.4->-r requirements.txt (line 4)) (2.8.2)\n","Requirement already satisfied: kiwisolver>=1.0.1 in /usr/local/lib/python3.7/dist-packages (from matplotlib==3.3.4->-r requirements.txt (line 4)) (1.4.4)\n","Requirement already satisfied: cycler>=0.10 in /usr/local/lib/python3.7/dist-packages (from matplotlib==3.3.4->-r requirements.txt (line 4)) (0.11.0)\n","Requirement already satisfied: pyyaml in /usr/local/lib/python3.7/dist-packages (from Keras==2.2.5->-r requirements.txt (line 5)) (6.0)\n","Requirement already satisfied: scipy>=0.14 in /usr/local/lib/python3.7/dist-packages (from Keras==2.2.5->-r requirements.txt (line 5)) (1.7.3)\n","Requirement already satisfied: typing-extensions in /usr/local/lib/python3.7/dist-packages (from kiwisolver>=1.0.1->matplotlib==3.3.4->-r requirements.txt (line 4)) (4.1.1)\n","Collecting scipy>=0.14\n","  Downloading scipy-1.7.2-cp37-cp37m-manylinux_2_12_x86_64.manylinux2010_x86_64.whl (38.2 MB)\n","  Downloading scipy-1.7.1-cp37-cp37m-manylinux_2_5_x86_64.manylinux1_x86_64.whl (28.5 MB)\n","  Downloading scipy-1.7.0-cp37-cp37m-manylinux_2_5_x86_64.manylinux1_x86_64.whl (28.5 MB)\n","  Downloading scipy-1.6.3-cp37-cp37m-manylinux1_x86_64.whl (27.4 MB)\n","  Downloading scipy-1.6.2-cp37-cp37m-manylinux1_x86_64.whl (27.4 MB)\n","  Downloading scipy-1.6.1-cp37-cp37m-manylinux1_x86_64.whl (27.4 MB)\n","  Downloading scipy-1.6.0-cp37-cp37m-manylinux1_x86_64.whl (27.4 MB)\n","  Downloading scipy-1.5.4-cp37-cp37m-manylinux1_x86_64.whl (25.9 MB)\n","Requirement already satisfied: setuptools>=41.0.0 in /usr/local/lib/python3.7/dist-packages (from tensorboard<1.15.0,>=1.14.0->tensorflow-gpu==1.14.0->-r requirements.txt (line 2)) (57.4.0)\n","Requirement already satisfied: werkzeug>=0.11.15 in /usr/local/lib/python3.7/dist-packages (from tensorboard<1.15.0,>=1.14.0->tensorflow-gpu==1.14.0->-r requirements.txt (line 2)) (1.0.1)\n","Requirement already satisfied: markdown>=2.6.8 in /usr/local/lib/python3.7/dist-packages (from tensorboard<1.15.0,>=1.14.0->tensorflow-gpu==1.14.0->-r requirements.txt (line 2)) (3.4.1)\n","Requirement already satisfied: importlib-metadata>=4.4 in /usr/local/lib/python3.7/dist-packages (from markdown>=2.6.8->tensorboard<1.15.0,>=1.14.0->tensorflow-gpu==1.14.0->-r requirements.txt (line 2)) (4.13.0)\n","Requirement already satisfied: zipp>=0.5 in /usr/local/lib/python3.7/dist-packages (from importlib-metadata>=4.4->markdown>=2.6.8->tensorboard<1.15.0,>=1.14.0->tensorflow-gpu==1.14.0->-r requirements.txt (line 2)) (3.10.0)\n","Requirement already satisfied: chardet<4,>=3.0.2 in /usr/local/lib/python3.7/dist-packages (from requests->-r requirements.txt (line 11)) (3.0.4)\n","Requirement already satisfied: idna<3,>=2.5 in /usr/local/lib/python3.7/dist-packages (from requests->-r requirements.txt (line 11)) (2.10)\n","Requirement already satisfied: certifi>=2017.4.17 in /usr/local/lib/python3.7/dist-packages (from requests->-r requirements.txt (line 11)) (2022.9.24)\n","Requirement already satisfied: urllib3!=1.25.0,!=1.25.1,<1.26,>=1.21.1 in /usr/local/lib/python3.7/dist-packages (from requests->-r requirements.txt (line 11)) (1.24.3)\n","Installing collected packages: numpy, h5py, tensorflow-estimator, tensorboard, scipy, Pillow, keras-applications, tqdm, tensorflow-gpu, opencv-python, matplotlib, Keras, imageio-ffmpeg, imageio\n","  Attempting uninstall: numpy\n","    Found existing installation: numpy 1.21.6\n","    Uninstalling numpy-1.21.6:\n","      Successfully uninstalled numpy-1.21.6\n","  Attempting uninstall: h5py\n","    Found existing installation: h5py 3.1.0\n","    Uninstalling h5py-3.1.0:\n","      Successfully uninstalled h5py-3.1.0\n","  Attempting uninstall: tensorflow-estimator\n","    Found existing installation: tensorflow-estimator 2.9.0\n","    Uninstalling tensorflow-estimator-2.9.0:\n","      Successfully uninstalled tensorflow-estimator-2.9.0\n","  Attempting uninstall: tensorboard\n","    Found existing installation: tensorboard 2.9.1\n","    Uninstalling tensorboard-2.9.1:\n","      Successfully uninstalled tensorboard-2.9.1\n","  Attempting uninstall: scipy\n","    Found existing installation: scipy 1.7.3\n","    Uninstalling scipy-1.7.3:\n","      Successfully uninstalled scipy-1.7.3\n","  Attempting uninstall: Pillow\n","    Found existing installation: Pillow 7.1.2\n","    Uninstalling Pillow-7.1.2:\n","      Successfully uninstalled Pillow-7.1.2\n","  Attempting uninstall: tqdm\n","    Found existing installation: tqdm 4.64.1\n","    Uninstalling tqdm-4.64.1:\n","      Successfully uninstalled tqdm-4.64.1\n","  Attempting uninstall: opencv-python\n","    Found existing installation: opencv-python 4.6.0.66\n","    Uninstalling opencv-python-4.6.0.66:\n","      Successfully uninstalled opencv-python-4.6.0.66\n","  Attempting uninstall: matplotlib\n","    Found existing installation: matplotlib 3.2.2\n","    Uninstalling matplotlib-3.2.2:\n","      Successfully uninstalled matplotlib-3.2.2\n","  Attempting uninstall: Keras\n","    Found existing installation: keras 2.9.0\n","    Uninstalling keras-2.9.0:\n","      Successfully uninstalled keras-2.9.0\n","  Attempting uninstall: imageio\n","    Found existing installation: imageio 2.9.0\n","    Uninstalling imageio-2.9.0:\n","      Successfully uninstalled imageio-2.9.0\n","Successfully installed Keras-2.2.5 Pillow-8.4.0 h5py-2.9.0 imageio-2.15.0 imageio-ffmpeg-0.4.7 keras-applications-1.0.8 matplotlib-3.3.4 numpy-1.16.4 opencv-python-4.5.3.56 scipy-1.5.4 tensorboard-1.14.0 tensorflow-estimator-1.14.0 tensorflow-gpu-1.14.0 tqdm-4.32.2\n"]},{"output_type":"stream","name":"stderr","text":["bash: line 1: conda: command not found\n","bash: line 2: conda: command not found\n","ERROR: pip's dependency resolver does not currently take into account all the packages that are installed. This behaviour is the source of the following dependency conflicts.\n","xarray 0.20.2 requires numpy>=1.18, but you have numpy 1.16.4 which is incompatible.\n","xarray-einstats 0.2.2 requires numpy>=1.21, but you have numpy 1.16.4 which is incompatible.\n","tensorflow 2.9.2 requires keras<2.10.0,>=2.9.0rc0, but you have keras 2.2.5 which is incompatible.\n","tensorflow 2.9.2 requires numpy>=1.20, but you have numpy 1.16.4 which is incompatible.\n","tensorflow 2.9.2 requires tensorboard<2.10,>=2.9, but you have tensorboard 1.14.0 which is incompatible.\n","tensorflow 2.9.2 requires tensorflow-estimator<2.10.0,>=2.9.0rc0, but you have tensorflow-estimator 1.14.0 which is incompatible.\n","tables 3.7.0 requires numpy>=1.19.0, but you have numpy 1.16.4 which is incompatible.\n","spacy 3.4.2 requires tqdm<5.0.0,>=4.38.0, but you have tqdm 4.32.2 which is incompatible.\n","scikit-image 0.18.3 requires numpy>=1.16.5, but you have numpy 1.16.4 which is incompatible.\n","resampy 0.4.2 requires numpy>=1.17, but you have numpy 1.16.4 which is incompatible.\n","pywavelets 1.3.0 requires numpy>=1.17.3, but you have numpy 1.16.4 which is incompatible.\n","pyerfa 2.0.0.1 requires numpy>=1.17, but you have numpy 1.16.4 which is incompatible.\n","pyarrow 6.0.1 requires numpy>=1.16.6, but you have numpy 1.16.4 which is incompatible.\n","prophet 1.1.1 requires tqdm>=4.36.1, but you have tqdm 4.32.2 which is incompatible.\n","plotnine 0.8.0 requires numpy>=1.19.0, but you have numpy 1.16.4 which is incompatible.\n","panel 0.12.1 requires tqdm>=4.48.0, but you have tqdm 4.32.2 which is incompatible.\n","pandas 1.3.5 requires numpy>=1.17.3; platform_machine != \"aarch64\" and platform_machine != \"arm64\" and python_version < \"3.10\", but you have numpy 1.16.4 which is incompatible.\n","pandas-gbq 0.17.9 requires numpy>=1.16.6, but you have numpy 1.16.4 which is incompatible.\n","numba 0.56.4 requires numpy<1.24,>=1.18, but you have numpy 1.16.4 which is incompatible.\n","kapre 0.3.7 requires numpy>=1.18.5, but you have numpy 1.16.4 which is incompatible.\n","jaxlib 0.3.22+cuda11.cudnn805 requires numpy>=1.20, but you have numpy 1.16.4 which is incompatible.\n","jax 0.3.23 requires numpy>=1.20, but you have numpy 1.16.4 which is incompatible.\n","gym 0.25.2 requires numpy>=1.18.0, but you have numpy 1.16.4 which is incompatible.\n","db-dtypes 1.0.4 requires numpy<2.0dev,>=1.16.6, but you have numpy 1.16.4 which is incompatible.\n","cmdstanpy 1.0.8 requires numpy>=1.21, but you have numpy 1.16.4 which is incompatible.\n","astropy 4.3.1 requires numpy>=1.17, but you have numpy 1.16.4 which is incompatible.\n","aesara 2.7.9 requires numpy>=1.17.0, but you have numpy 1.16.4 which is incompatible.\n","aeppl 0.0.33 requires numpy>=1.18.1, but you have numpy 1.16.4 which is incompatible.\n"]}]},{"cell_type":"code","source":["%cd Deep-Generative-Projection/\n","!unzip pretrained_models.zip"],"metadata":{"colab":{"base_uri":"https://localhost:8080/"},"id":"OCfchH_r3-h0","executionInfo":{"status":"ok","timestamp":1668449638194,"user_tz":300,"elapsed":12809,"user":{"displayName":"Francesco Bassino","userId":"12214195385968794219"}},"outputId":"8ce7379f-3734-4d3a-f2b5-3e9a6f76734c"},"execution_count":3,"outputs":[{"output_type":"stream","name":"stdout","text":["/content/Deep-Generative-Projection\n","Archive:  pretrained_models.zip\n","   creating: pretrained_models/\n","  inflating: pretrained_models/attributes_model_resnet101.h5  \n","  inflating: pretrained_models/encoder-stylegan2-upper-body-512.pkl  \n","  inflating: pretrained_models/vgg.h5  \n"]}]},{"cell_type":"code","source":["!python deepGenPro.py --cloth_dir=./data/model_1/cloth_1/ --model_dir=./data/model_1/model_info/ --cloth_sleeve=short --output_dir=./output/"],"metadata":{"colab":{"base_uri":"https://localhost:8080/"},"id":"Ucz6IIzTWTI9","executionInfo":{"status":"ok","timestamp":1668449665220,"user_tz":300,"elapsed":4297,"user":{"displayName":"Francesco Bassino","userId":"12214195385968794219"}},"outputId":"a49c3651-f7e1-451d-d126-3f43e74e2421"},"execution_count":5,"outputs":[{"output_type":"stream","name":"stdout","text":["Traceback (most recent call last):\n","  File \"deepGenPro.py\", line 4, in <module>\n","    import tensorflow as tf\n","  File \"/usr/local/lib/python3.7/dist-packages/tensorflow/__init__.py\", line 28, in <module>\n","    from tensorflow.python import pywrap_tensorflow  # pylint: disable=unused-import\n","  File \"/usr/local/lib/python3.7/dist-packages/tensorflow/python/__init__.py\", line 83, in <module>\n","    from tensorflow.python import keras\n","  File \"/usr/local/lib/python3.7/dist-packages/tensorflow/python/keras/__init__.py\", line 26, in <module>\n","    from tensorflow.python.keras import activations\n","  File \"/usr/local/lib/python3.7/dist-packages/tensorflow/python/keras/activations.py\", line 24, in <module>\n","    from tensorflow.python.keras.utils.generic_utils import deserialize_keras_object\n","  File \"/usr/local/lib/python3.7/dist-packages/tensorflow/python/keras/utils/__init__.py\", line 39, in <module>\n","    from tensorflow.python.keras.utils.multi_gpu_utils import multi_gpu_model\n","  File \"/usr/local/lib/python3.7/dist-packages/tensorflow/python/keras/utils/multi_gpu_utils.py\", line 22, in <module>\n","    from tensorflow.python.keras.engine.training import Model\n","  File \"/usr/local/lib/python3.7/dist-packages/tensorflow/python/keras/engine/training.py\", line 40, in <module>\n","    from tensorflow.python.keras.engine import network\n","  File \"/usr/local/lib/python3.7/dist-packages/tensorflow/python/keras/engine/network.py\", line 39, in <module>\n","    from tensorflow.python.keras import saving\n","  File \"/usr/local/lib/python3.7/dist-packages/tensorflow/python/keras/saving/__init__.py\", line 33, in <module>\n","    from tensorflow.python.keras.saving.saved_model import export_saved_model\n","ImportError: cannot import name 'export_saved_model' from 'tensorflow.python.keras.saving.saved_model' (/usr/local/lib/python3.7/dist-packages/tensorflow/python/keras/saving/saved_model/__init__.py)\n"]}]}]}